explorer/lib/explorer_web/live/utils.ex

```diff
@@ -167,21 +172,36 @@ defmodule Utils do
     |> Enum.reverse()
   end
 
-  def calculate_proof_hashes({:ok, deserialized_batch}) do
+  def calculate_proof_hashes(deserialized_batch) do
     deserialized_batch
     |> Enum.map(fn s3_object ->
       :crypto.hash(:sha3_256, s3_object["proof"])
     end)
   end
 
-  def calculate_proof_hashes({:error, reason}) do
-    Logger.error("Error calculating proof hashes: #{inspect(reason)}")
-    []
+  defp stream_handler({:headers, _headers}, acc), do: {:cont, acc}
+  defp stream_handler({:status, 200}, acc), do: {:cont, acc}
+
+  defp stream_handler({:status, status_code}, _acc),
+    do: {:halt, {:error, {:http_error, status_code}}}
+
+  defp stream_handler({:data, chunk}, {_acc_body, size})
+       when size + byte_size(chunk) > @max_batch_size do
+    {:halt, {:error, {:http, :body_too_large}}}
+  end
+
+  defp stream_handler({:data, chunk}, {acc_body, size}) do
+    new_size = size + byte_size(chunk)
+    {:cont, {acc_body <> chunk, new_size}}
```

Comment:
instead of separating the function in these two cases, I think it is a bit more understandable to condense these in one function and check the condition `size + byte_size(chunk) > @max_batch_size` inside it

Reply:
also, for consistency I think it is better to rename `{acc_body, size}` to `{acc_body, acc_size}`

Reply:
Done!